File: src/main/java/net/softbell/bsh/dto/view/ActionInfoCardDto.java
package net.softbell.bsh.dto.view;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.EnableStatusRule;
import net.softbell.bsh.domain.entity.NodeAction;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 예약 등록 및 수정뷰 정보 카드정보 DTO
 */
@Getter
@Setter
public class ActionInfoCardDto
{
	private long actionId;
	private boolean enableStatus;
	private String description;
	
	public ActionInfoCardDto(NodeAction entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Init
		this.enableStatus = false;
		
		// Convert
		this.actionId = entity.getActionId();
		if (entity.getEnableStatus() == EnableStatusRule.ENABLE)
			this.enableStatus = true;
		this.description = entity.getDescription();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/general/ActionInfoCardDto.kt
package net.softbell.bsh.dto.view.general

import net.softbell.bsh.domain.EnableStatusRule
import net.softbell.bsh.domain.entity.NodeAction

/**
 * @author : Bell(bell@softbell.net)
 * @description : 예약 등록 및 수정뷰 정보 카드정보 DTO
 */
class ActionInfoCardDto(entity: NodeAction) {
    val actionId: Long = entity.actionId
    val enableStatus: Boolean = entity.enableStatus == EnableStatusRule.ENABLE
    val description: String = entity.description
}